[website] Add careers page for intern (#26280)
mui-org/material-ui: docs/src/pages/company/careers/careers.md

We are looking for senior software engineers to help support our open-source components, assist the Material-UI community, and grow our enterprise edition.

### [Software Engineer Intern](/company/software-engineer-intern/)

> We are looking for software engineer intern to help support our open-source components, assist the Material-UI community, and help us build new components.

Reviewer: ```suggestion
We're looking for software engineering intern to help support our open-source library, assist the Material-UI community, and help us build new components.
```
